flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/DorisWriteMetrics.java

```diff
@@ -0,0 +1,383 @@
+        streamLoadPutTimeHistogramMs =
+                sinkMetricGroup.histogram(
+                        String.format(
+                                METRIC_NAME_FORMAT,
+                                tableIdentifier,
+                                subTaskId,
+                                HISTOGRAM_STREAM_LOAD_PUT_DATA_TIME_MS),
+                        new DescriptiveStatisticsHistogram(HISTOGRAM_WINDOW_SIZE));
+        commitAndPublishTimeHistogramMs =
+                sinkMetricGroup.histogram(
+                        String.format(
+                                METRIC_NAME_FORMAT,
+                                tableIdentifier,
+                                subTaskId,
+                                HISTOGRAM_COMMIT_AND_PUBLISH_TIME_MS),
+                        new DescriptiveStatisticsHistogram(HISTOGRAM_WINDOW_SIZE));
+        this.beginTxnTimeHistogramMs =
+                sinkMetricGroup.histogram(
+                        String.format(
+                                METRIC_NAME_FORMAT,
+                                tableIdentifier,
+                                subTaskId,
+                                HISTOGRAM_BEGIN_TXN_TIME_MS),
+                        new DescriptiveStatisticsHistogram(HISTOGRAM_WINDOW_SIZE));
+        readDataTimeHistogramMs =
+                sinkMetricGroup.histogram(
+                        String.format(
+                                METRIC_NAME_FORMAT,
+                                tableIdentifier,
+                                subTaskId,
+                                HISTOGRAM_READ_DATA_TIME_MS),
+                        new DescriptiveStatisticsHistogram(HISTOGRAM_WINDOW_SIZE));
+        writeDataTimeHistogramMs =
+                sinkMetricGroup.histogram(
+                        String.format(
+                                METRIC_NAME_FORMAT,
+                                tableIdentifier,
+                                subTaskId,
+                                HISTOGRAM_WRITE_DATA_TIME_MS),
+                        new DescriptiveStatisticsHistogram(HISTOGRAM_WINDOW_SIZE));
+    }
+
+    private void register(SinkWriterMetricGroup sinkMetricGroup, int subTaskId) {
+        register(sinkMetricGroup, subTaskId, HISTOGRAM_WINDOW_SIZE);
+    }
+
+    private void flushSuccessLoad(RespContent responseContent) {
+        Optional.ofNullable(responseContent.getLoadBytes()).ifPresent(totalFlushLoadBytes::inc);
+        Optional.ofNullable(responseContent.getNumberLoadedRows())
+                .ifPresent(totalFlushLoadedRows::inc);
+        Optional.ofNullable(responseContent.getNumberTotalRows())
+                .ifPresent(totalFlushNumberTotalRows::inc);
+        Optional.ofNullable(responseContent.getNumberFilteredRows())
+                .ifPresent(totalFlushFilteredRows::inc);
+        Optional.ofNullable(responseContent.getLoadTimeMs()).ifPresent(totalFlushTimeMs::inc);
```

Comment:
Is getLoadTimeMs also a counter-type metric?

Reply:
There are two  metric type for LoadTimeMs:  counter and Historgram ,